apollo-router/src/plugin/mod.rs

```diff
@@ -192,19 +195,18 @@ pub trait Plugin: Send + Sync + 'static {
         service
     }
 
-    /// The `custom_endpoint` method lets you declare a new endpoint exposed for your plugin.
-    /// For now it's only accessible for official `apollo.` plugins and for `experimental.`. This endpoint will be accessible via `/plugins/group.plugin_name`
-    fn custom_endpoint(&self) -> Option<transport::BoxService> {
-        None
-    }
-
     /// Return the name of the plugin.
     fn name(&self) -> &'static str
     where
         Self: Sized,
     {
         get_type_of(self)
     }
+
+    /// Return one or several `Endpoint`s and `ListenAddr` and the router will serve your custom web Endpoint(s).
```

Comment:
Are we going to remove this from the public API?
I'm still uneasy about us abstracting away axum::Router via the endpoint struct.
Maybe this is why: https://github.com/apollographql/router/pull/1654/files#diff-37adf9e170c9b384f17336e5b5e5bf9cd94fd1d618b8969996a5ad56b635ace6R483-R516 is looking so complex?

Reply:
As decided together, we have added a documentation that mentions this method is experimental as of yet in ffc07a0